src/riot-rs-embassy/src/lib.rs

```diff
@@ -93,15 +108,23 @@ pub(crate) fn init() {
 #[cfg(feature = "executor-single-thread")]
 #[export_name = "riot_rs_embassy_init"]
 fn init() -> ! {
-    println!("riot-rs-embassy::init()");
+    println!("riot-rs-embassy::init(): using single thread executor");
     let p = arch::init();
 
-    println!("riot-rs-embassy::init() done");
-
     let executor = make_static!(arch::Executor::new());
     executor.run(|spawner| spawner.must_spawn(init_task(p)));
 }
 
+#[cfg(feature = "executor-thread")]
+#[riot_rs_macros::thread(autostart, stacksize = 65536, priority = 8)]
```

Comment:
There are some magic numbers here. 

I wonder if we should maybe be more explicit about this thread.
We could make the `init` function public, have it never return and make it `#[must_use]`. So with the `executor-thread` feature enabled, the user would have to create one thread where embassy is initialized and the tasks run. That way they would also have control over the priority and stacksize.

Reply:
> There are some magic numbers here.

yup, these are my main blocker for this PR. :(

I'd rather keep this inside riot-rs (not have the user start this). If we make the user start this, the application is suddenly tied to the thread mode executor. The user also has little information on how large the stack size should be. We hopefully have more insight (b/c we will gather information on our components, per-platform, and go with a `MAX (some, of, our, components, and,what, user, wants)`.

Reply:
> the application is suddenly tied to the thread mode executor

Isn't the application tied to the thread mode executor anyway as soon as they are using threads and embassy tasks together?

>  The user also has little information on how large the stack size should be.

They have some more info because they know how many tasks will run in it and their implementation. We ask them to define it for all their other threads, so why not here as well? 
And again, IMO the priority should definitely be defined by the user.

Reply:
Out-of-band discussion: Let's make it opt-in to explicitly created the thread in the user application. Default is the current version where `riot-rs-embassy` creates it.

Reply:
> Let's make it opt-in to explicitly created the thread in the user application.

I think we should postpone this, and allow a fully custom executor setup. We'd need to make `init_task()` and `arch::init()` public, and IMO there's a lot around that that needs figuring out (and getting documented).

Ok?

Reply:
I am okay with postponing it. 

Just wondering: why isn't it sufficient to just make `init` public? https://github.com/future-proof-iot/RIOT-rs/blob/220606c55bccaad1143a3e1001ba87b85eecb76d/src/riot-rs-embassy/src/lib.rs#L136-L139

Reply:
> Just wondering: why isn't it sufficient to just make `init` public (and `cfg_attr` the thread macro)?

Actually it might be. Then we'd need to make autostart feature gated, and come up with good names. Let's do that in a folo!